docs/integrations/destinations/weaviate.md

```diff
@@ -1,80 +1,88 @@
-ID will automatically be converted to a UUID. Weaviate only supports the ID to be a UUID.
-For example, if the record has `id=1` then this would become a uuid of
-`00000000-0000-0000-0000-000000000001`.
+When specifying text fields, you can access nested fields in the record by using dot notation, e.g. `user.name` will access the `name` field in the `user` object. It's also possible to use wildcards to access all fields in an object, e.g. `users.*.name` will access all `names` fields in all entries of the `users` array.
 
-Any field name starting with an upper case letter will be converted to lower case. For example,
-if you have a field name `USD` then that field will become `uSD`. This is due to a limitation
-in Weaviate, see [this issue in Weaviate](https://github.com/semi-technologies/weaviate/issues/2438).
+The chunk length is measured in tokens produced by the `tiktoken` library. The maximum is 8191 tokens, which is the maximum length supported by the `text-embedding-ada-002` model.
 
-## Getting Started
+The stream name gets added as a metadata field `_ab_stream` to each document. If available, the primary key of the record is used to identify the document to avoid duplications when updated versions of records are indexed. It is added as the `_ab_record_id` metadata field.
 
-Airbyte Cloud only supports connecting to your Weaviate Instance instance with TLS encryption and with a username and
-password.
+### Embedding
 
-## Getting Started \(Airbyte Open-Source\)
+The connector can use one of the following embedding methods:
 
-#### Requirements
-
-To use the Weaviate destination, you'll need:
+1. OpenAI - using [OpenAI API](https://beta.openai.com/docs/api-reference/text-embedding) , the connector will produce embeddings using the `text-embedding-ada-002` model with **1536 dimensions**. This integration will be constrained by the [speed of the OpenAI embedding API](https://platform.openai.com/docs/guides/rate-limits/overview).
 
-- A Weaviate cluster version 21.8.10.19 or above
+2. Cohere - using the [Cohere API](https://docs.cohere.com/reference/embed), the connector will produce embeddings using the `embed-english-light-v2.0` model with **1024 dimensions**. 
 
-#### Configure Network Access
+3. From field - if you have pre-calculated embeddings stored in a field in your source database, you can use the `From field` integration to load them into Weaviate. The field must be a JSON array of numbers, e.g. `[0.1, 0.2, 0.3]`. The array must have the same length as the embedding dimensionality of the index.
 
-Make sure your Weaviate database can be accessed by Airbyte. If your database is within a VPC, you may need to allow access from the IP you're using to expose Airbyte.
+4. No embedding - if you don't want to use embeddings or have configured a [vectorizer](https://weaviate.io/developers/weaviate/modules/retriever-vectorizer-modules) for your class, you can use the `No embedding` integration. This will store the text as-is in Weaviate.
```

Comment:
I believe the user-facing name of this is "Weaviate vectorizer" - should we align the docs with this name?